### PR TITLE
Add CapitalizeSelection command

### PR DESCRIPTION
**Pull request by codetheweb**
As requested by @kpathakota and @kevinslin. 😃

### CODE REVIEW

File: packages/plugin-core/src/commands/CapitalizeSelectionCommand.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  }
+  async execute() {
+    const maybeTextEditor = VSCodeUtils.getActiveTextEditor();
+
+    if (maybeTextEditor) {
+      const text = maybeTextEditor.document.getText(maybeTextEditor.selection);
+
+      const transformedText = text
+        .split(" ")
+        .map((token) => `${token[0].toUpperCase()}${token.slice(1)}`)
```
Comment (kevinslin) [inline packages/plugin-core/src/commands/CapitalizeSelectionCommand.ts:27]: What would happen for unicode input? eg. 🌱

File: packages/plugin-core/src/test/suite-integ/CapitalizeSelectionCommand.test.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+          body: "hello world",
+        });
+      },
+      onInit: async ({ vaults, wsRoot }) => {
+        const notePath = path.join(wsRoot, vaults[0].fsPath, "gamma.md");
+        const editor = await VSCodeUtils.openFileInEditor(
+          vscode.Uri.file(notePath)
+        );
+
+        if (!editor) {
```
Comment (kevinslin) [inline packages/plugin-core/src/test/suite-integ/CapitalizeSelectionCommand.test.ts:32]: since this is a test, we can just assert editor using an `as` operator

File: packages/plugin-core/src/commands/CapitalizeSelectionCommand.ts (hunk 1 above)
Comment (codetheweb) [reply packages/plugin-core/src/commands/CapitalizeSelectionCommand.ts]: It's generally unicode safe, i.e.:

```
> '👍'.toUpperCase();
'👍'
```

It does look like there are [one or two very minor edge cases](https://stackoverflow.com/a/30763402/2129808) when dealing with other languages, so I changed it to toLocaleUpperCase() instead to be on the safe side: 2f3929be

File: packages/plugin-core/src/test/suite-integ/CapitalizeSelectionCommand.test.ts (hunk 2 above)
Comment (codetheweb) [reply packages/plugin-core/src/test/suite-integ/CapitalizeSelectionCommand.test.ts]: 1949b152

**Comment by codetheweb** [comment]
It's unclear why that one test is failing; I don't think it's related to anything I changed.
Says the snapshot is different but only for Node 14 on macOS which seems very odd.

**Comment by kevinslin** [comment]
That test is known to be flaky. I need to change the snapshot so no worries.
